api/agent/agent.go

```diff
@@ -737,114 +745,183 @@ func newSizerRespWriter(max uint64, rw http.ResponseWriter) http.ResponseWriter
 
 func (s *sizerRespWriter) Write(b []byte) (int, error) { return s.w.Write(b) }
 
-func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state ContainerState) {
+func (a *agent) waitInitialize(ctx context.Context, caller slotCaller, evictor *EvictToken, udsWait <-chan error) (bool, error) {
+	for {
+		select {
+		case err := <-udsWait:
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		case <-ctx.Done(): // container shutdown
+			return false, nil
+		case <-a.shutWg.Closer(): // agent shutdown
+			return false, nil
+		case <-caller.done: // original caller disconnected or serviced by another container?
+			evictor.SetEvictable(true)
+			caller.done = nil // block 'doneChan' after this point
+		case <-evictor.C: // eviction
+			return false, nil
+		}
+	}
+}
+
+// Try to queue an error to the error channel if possible.
+func tryQueueErr(err error, ch chan error) error {
+	if err != nil {
+		select {
+		case ch <- err:
+		default:
+		}
+	}
+	return err
+}
+
+func (a *agent) runHot(ctx context.Context, caller slotCaller, call *call, tok ResourceToken, state ContainerState) {
```

Comment:
would it help to return an `err` from `runHot` and have the calling thread handle queueing the error to the slot queue? idk, idea, there's a lot of queueing errors so we'd go to more natural return err and it would be one less thing in the defer. i guess i see 2 uses of tryQueueError from a child goroutine. leaving idea, but feel free to ignore

Reply:
I agree, the queue error logic needs rework. But saving that for another PR.

Reply:
err handling needs a rework, but I'm deferring that to a future PR.